client/signup/main.jsx

```diff
@@ -329,7 +330,32 @@ const Signup = React.createClass( {
 		// redirect the user to the next step
 		scrollPromise.then( () => {
 			if ( ! this.isEveryStepSubmitted() ) {
-				page( utils.getStepUrl( this.props.flowName, stepName, stepSectionName, this.props.locale ) );
+				if ( 'skip' === abtest( 'skipThemesSelectionModal' ) && 'themes' === stepName ) {
+					const flowSteps = flows.getFlow( this.props.flowName, stepName ).steps,
+						currentStepIndex = indexOf( flowSteps, stepName ),
+						nextStepName = flowSteps[ currentStepIndex + 1 ],
+						nextProgressItem = this.state.progress[ currentStepIndex + 1 ],
+						nextStepSection = nextProgressItem && nextProgressItem.stepSectionName || '';
+
+					const designType = this.props.signupDependencies && this.props.signupDependencies.designType;
+
+					if ( designType ) {
```

Comment:
We will usually be coming from the _Design Type_ selection screen, which sets `designType` on the `signupDependencies`. Maybe I should remove the `if` block and just explicitly let it be known that `designType` is expected to be set at this stage?